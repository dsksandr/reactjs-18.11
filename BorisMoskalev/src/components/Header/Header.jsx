import React, {Component} from 'react';
import Toolbar from '@material-ui/core/Toolbar';
import {Link} from 'react-router-dom';

export class Header extends Component {
    render() {
        return (
            <Toolbar>
                <Link to='/'> Chats </Link>
                <Link to='/profile/'>Profile</Link>
            </Toolbar>
        )
    }
}