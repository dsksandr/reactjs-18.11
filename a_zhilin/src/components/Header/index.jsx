import './style.scss'

import React from 'react'
import { Link } from 'react-router-dom'
import {
  AppBar,
  Button,
  IconButton,
  Toolbar,
  Typography
} from '@material-ui/core/'
import MenuIcon from '@material-ui/icons/Menu'

import useStyles from './style'

const Header = () => {
  /**
   * @type object classes
   */
  const classes = useStyles()

  return (
    <div className={classes.root} id='header'>
      <AppBar position='static'>
        <Toolbar>
          <IconButton
            edge='start'
            className={classes.menuButton}
            color='inherit'
            aria-label='menu'
          >
            <MenuIcon />
          </IconButton>
          <Link to='/' className='header__logo'>
            <Typography variant='h6' className={classes.title}>
              MyChat
            </Typography>
          </Link>
          <Button color='inherit'>Login</Button>
        </Toolbar>
      </AppBar>
    </div>
  )
}

export default Header
