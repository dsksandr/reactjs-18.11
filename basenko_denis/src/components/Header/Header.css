.header{
   background-color: indigo;
   padding: 0 20px;
   height: 80px;
   display: flex;
   align-items: center;
}
.header__a_profile{
    border-radius: 60px;
    height: 60px;
    width: 60px;
    border: 2px solid yellow;
    color: yellow;
    display: block;
    text-align: center;
    line-height: 60px;
}