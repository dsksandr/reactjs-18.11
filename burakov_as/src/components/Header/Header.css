.root {
  flexGrow: 1;
  background-color: grey;
}

.paper {
  padding: 15px;
  textAlign: 'center';
  background-color: grey;
}