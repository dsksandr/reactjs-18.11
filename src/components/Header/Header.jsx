import React, { PureComponent } from 'react';

export class Header extends PureComponent {
  render() {
    const { chats } = this.props;

    return (
      <header>
        Hi from header!
        {
          chats && chats.length && <ul>
            {chats.map((chat, idx) => <li key={idx}>{chat.name}</li>)}
          </ul>
        }
      </header>
    );
  }
}