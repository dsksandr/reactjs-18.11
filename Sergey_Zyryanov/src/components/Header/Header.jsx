import React, {Component} from "react";
import { Link } from 'react-router-dom';

import './Header.scss';
import Button from "@material-ui/core/Button";

export class Header extends Component{
  render() {
    return (
      <div className='header'>
        <Link to="/profile">
          <Button variant="contained" color="secondary">
            Профайл
          </Button>
        </Link>
        <Link to="/about">
          <Button variant="contained" color="primary">
            About
          </Button>
        </Link>
      </div>
    );
  }
}