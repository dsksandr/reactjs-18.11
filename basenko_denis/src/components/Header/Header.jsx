import './Header.css'

import React from 'react'
import { Link } from 'react-router-dom'

export class Header extends React.Component{

    render(){
        return (
            <header className='header'>
                <Link className='header__a_profile' to='/profile'>Profile</Link>
            </header>
        )
    }

}