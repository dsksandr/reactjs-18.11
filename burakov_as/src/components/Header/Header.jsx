import './Header.css';

import React, { PureComponent } from 'react';
import { Link } from 'react-router-dom';
import Button from '@material-ui/core/Button';
import Paper from '@material-ui/core/Paper';
import Grid from '@material-ui/core/Grid';

export class Header extends PureComponent{
  render() {
    const { chats } = this.props;

    return (
      <div>
        <header>
          {
            chats && chats.length && <ul>
              {chats.map((chat, idx) => <li key={idx}>{chat.name}</li>)}
            </ul>
          }
        </header>

        <Grid container>
          <Grid item xs={12}>
            <Paper className="paper">
            
              <Link className="no-decor" to="/profile">  
              <Button variant="contained" color="primary">
                Profile
              </Button>
              </Link>
            </Paper>   
          </Grid>
        </Grid>
      </div>
    )
  }
}