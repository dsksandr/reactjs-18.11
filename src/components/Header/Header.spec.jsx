import React from 'react';
import { Header } from './Header';

import renderer from 'react-test-renderer';
import { shallow } from 'enzyme';

test('Header should be rendered', () => { // snapshot test
  const tree = renderer
    .create(<Header chats={[{name: 'Chat1'}, {name: 'Chat2'}]} />)
    .toJSON();

  expect(tree).toMatchSnapshot();
});

test('Header should be rendered as ul', () => { // shallow test
  const header = shallow(<Header chats={[{name: 'Chat1'}, {name: 'Chat2'}]} />);
  const lis = header.find('li');

  expect(lis.length).toBe(2);
  expect(lis.first().text()).toBe('Chat1');
  expect(lis.last().text()).toBe('Chat2');
});