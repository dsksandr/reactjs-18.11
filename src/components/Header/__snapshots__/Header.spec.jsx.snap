// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Header should be rendered 1`] = `
<header>
  Hi from header!
  <ul>
    <li>
      Chat1
    </li>
    <li>
      Chat2
    </li>
  </ul>
</header>
`;
